### PR TITLE
Expose run statistics to optimisation wrappers

### PR DESCRIPTION
**Pull request by jetuk**
Log the run time and speed during optimisation, and also save the `ModelResult` on to the wrapper so it can be inspected if required.

### CODE REVIEW

**Comment by jetuk** [comment]
The failure here is just appveyor being a bit terrible.

File: pywr/optimisation/platypus.py [hunk 1]
```diff
@@ -106,7 +106,7 @@ def evaluate(self, solution):
                 ints = np.round(np.array(x[-var.integer_size:])).astype(np.int32)
                 var.set_integer_variables(ints)
 
-        run_stats = self.model.run()
+        self.run_stats = self.model.run()
```
Comment (Batch21) [inline pywr/optimisation/platypus.py:109]: Won't this attribute get continuously overwritten during the optimisation?

Comment (jetuk) [reply pywr/optimisation/platypus.py:109]: Yes, but this allows you to access the run statistics if you subclass the wrapper. E.g. if you want to save them somewhere.

Comment (Batch21) [reply pywr/optimisation/platypus.py:109]: ah ok

**Pull request merged**